app/controllers/application_controller.rb

```diff
@@ -24,4 +27,11 @@ def bad_request
   def unauthorized
     head :unauthorized
   end
+
+  def switch_locale(&action)
+    settings = SettingsHelper::Settings.new(cookies)
```

Comment:
Also factor in what's in the `Accept-Language` header, if anything.

Reply:
Done